app/src/main/java/org/mozilla/fenix/compose/cfr/CFRPopupContent.kt

```diff
@@ -0,0 +1,171 @@
+    popupWidth: Dp = CFRPopup.DEFAULT_WIDTH.dp,
+    action: @Composable (() -> Unit) = {}
+) {
+    val popupShape = CFRPopupShape(
+        indicatorDirection,
+        indicatorArrowStartOffset,
+        CFRPopup.DEFAULT_INDICATOR_HEIGHT.dp,
+        CFRPopup.DEFAULT_CORNER_RADIUS.dp,
+    )
+
+    Box(modifier = Modifier.width(popupWidth + CFRPopup.DEFAULT_HORIZONTAL_PADDING.dp * 2)) {
+        Surface(
+            color = Color.Transparent,
+            // Need to override the default RectangleShape to avoid casting shadows for that shape.
+            shape = popupShape,
+            modifier = Modifier
+                .align(Alignment.Center)
+                .background(
+                    shape = popupShape,
+                    brush = Brush.linearGradient(
+                        colors = listOf(
+                            FirefoxTheme.colors.gradientEnd,
+                            FirefoxTheme.colors.gradientStart
+                        ),
+                        end = Offset(0f, Float.POSITIVE_INFINITY),
+                        start = Offset(Float.POSITIVE_INFINITY, 0f)
+                    )
+                )
+                .wrapContentHeight()
+                .width(popupWidth)
+        ) {
+
+            Column(
+                modifier = Modifier
+                    .padding(
+                        start = 16.dp,
+                        top = 16.dp + if (indicatorDirection == CFRPopup.IndicatorDirection.UP) {
+                            CFRPopup.DEFAULT_INDICATOR_HEIGHT.dp
+                        } else {
+                            0.dp
+                        },
+                        end = 16.dp,
+                        bottom = 16.dp +
+                            if (indicatorDirection == CFRPopup.IndicatorDirection.DOWN) {
+                                CFRPopup.DEFAULT_INDICATOR_HEIGHT.dp
+                            } else {
+                                0.dp
+                            }
+                    )
+            ) {
+                Text(
+                    text = text,
+                    modifier = Modifier.padding(
+                        end = 24.dp, // 8.dp extra padding to the "X" icon
+                    ),
+                    color = FirefoxTheme.colors.textOnColorPrimary,
+                    style = FirefoxTheme.typography.body2
+                )
+
+                action()
+            }
+        }
+
+        IconButton(
+            onClick = { onDismiss(true) },
+            modifier = Modifier
+                .align(Alignment.TopEnd)
+                .padding(
+                    top = if (indicatorDirection == CFRPopup.IndicatorDirection.UP) 14.dp else 0.dp,
+                    end = 6.dp
+                )
+                .size(48.dp)
+        ) {
+            Icon(
+                imageVector = Filled.Close,
+                contentDescription = stringResource(string.cfr_dismiss_button_default_content_description),
```

Comment:
Was the `R.string` imported?
```suggestion
                contentDescription = stringResource(R.string.cfr_dismiss_button_default_content_description),
```

Reply:
Yes, thank you!
Move some code around and imports got mangled, also added some `R.`s, will check again for all.